src/nvim/eval.c

```diff
@@ -12716,11 +12716,11 @@ static void f_jobstop(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     // Ignore return code, but show error later.
     (void)channel_close(data->id, kChannelPartRpc, &error);
   }
-  process_stop((Process *)&data->stream.proc);
-  rettv->vval.v_number = 1;
+  bool valid_proc = process_stop((Process *)&data->stream.proc);
```

Comment:
This covers the case where the job-id is valid but the process already stopped.

Do we also need to cover the case of an invalid job-id? I.e. pass `false` to `find_job()` as done in #10984 ?

Reply:
I do not quite understand why invalid job-id could be handled when passing `false` to `find_job()`, cause `EMSG` macro will return `true`?

Reply:
`find_job(...,false)` will return NULL if job-id is invalid. Then `f_jobstop` should return `-1`.

Does that make sense?

Reply:
From your suggestion
* Remove the modification in process_stop(), using find_job(..., false) to check the valid job id
* Return -1 for invalid id, 1 for valid id (or you think valid id should return 0)

BTW, as changing to find_job(..., false) there are some functional tests failed, cause the previous test will throw out an error when the job id is non-existent or is stopped already.
I should change them to check the return value depending on this PR, right?

Reply:
Do we really need to do this? a job id will become invalid "soon" after the job have been stopped (after one or two uv ticks), so it is that meaningful to distinguish the two? I think this should just return 1 if the job was stopped and zero otherwise (the job was just stopped, or exited a second ago, or five minutes ago).

Reply:
Fine with me, but...

> a job id will become invalid "soon" after the job have been stopped (after one or two uv ticks)

my understanding of https://github.com/neovim/neovim/issues/10943 was that we don't want jobstop() to fail loudly even if the job-id is now invalid. Because some plugin might hang onto some old job-id too long, but it shouldn't need to try/catch it. However, maybe this is too "sloppy" and the common case is already addressed by this PR.

Reply:
@justinmk but that is 100% consistent with what I suggested. It should return zero I said. I did not say that it should throw an error.